lib/mutant/isolation/fork.rb

```diff
@@ -0,0 +1,72 @@
+module Mutant
+  class Isolation
+    # Isolation via the fork(2) systemcall.
+    #
+    # We do inject so many globals and common patterns to make this unit
+    # specifiable without mocking the globals and more important: Not having
+    # mutations that bypass mocks into a real world side effect.
+    class Fork < self
+      include Anima.new(:process, :stderr, :stdout, :io, :devnull, :marshal)
+
+      # Prevent mutation from process.fork in #call to call Kernel#fork
+      undef_method :fork
+
+      # Call block in isolation
+      #
+      # @return [Object]
+      #   returns block execution result
+      #
+      # @raise [Error]
+      #   if block terminates abnormal
+      def call(&block)
+        io.pipe(binmode: true) do |reader, writer|
+          parent(reader, writer, &block)
```

Comment:
> This would not enforce arity.

The `#parent` method enforces arity though.

This is what I was proposing:

```ruby
def call(&block)
  io.pipe(binmode: true) do |*pair|
    parent(*pair, &block)
  end
rescue => exception
  raise Error, exception
end
```